### PR TITLE
MNTOR-3219 - sendEmail function should logs sends and errors

### PR DESCRIPTION
**Pull request by rhelmer**
# References: 
Jira: MNTOR-3219

# Description

The `sendEmail` function doesn't currently log sends and errors on its own, this leaves it up to the caller but it's an important function that we should be logging. It made some recent SES configuration changes more difficult to track down than we would've liked.

# How to test

Sending email locally or on stage/prod should now log when an email is sent or an error occurs, in addition to returning info to the caller.

# Checklist (Definition of Done)
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] I've added a unit test to test for potential regressions of this bug.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/utils/email.js [hunk 1]
```diff
@@ -51,40 +51,45 @@ async function initEmail (smtpUrl = AppConstants.SMTP_URL) {
  * @param {string} recipient
  * @param {string} subject
  * @param {string} html
- * @returns {Promise<SMTPTransport.SentMessageInfo>} <Promise>
+ *
+ * @returns {Promise<SMTPTransport.SentMessageInfo>}
  */
-function sendEmail (recipient, subject, html) {
+async function sendEmail (recipient, subject, html) {
```
Comment (rhelmer) [inline src/utils/email.js:57]: I also turned this into an `async function` so we don't need to do quite so much wrapping.

File: src/utils/email.js [hunk 2]
```diff
@@ -51,40 +51,45 @@ async function initEmail (smtpUrl = AppConstants.SMTP_URL) {
...
-      /* c8 ignore next 4 */
-      if (gTransporter.transporter.name === 'JSONTransport') {
-        // @ts-ignore Added typing later, but it disagrees with actual use:
-        console.info('JSONTransport', { message: info.message.toString() })
-      }
-      resolve(info)
-    })
-  })
+  try {
+    const info = await gTransporter.sendMail(mailOptions);
```
Comment (rhelmer) [inline src/utils/email.js:74]: `.sendMail()` returns a `Promise` if called without the callback function.

**Comment by rhelmer** [comment]
It would be really nice to convert this to typescript, but I wanted this change to be a bit more surgical.

**Review by Vinnl — APPROVED** [review]
I didn't see any errors logged locally even though AFAIK emails still don't get sent, but I think the problem there was that it didn't even produce an error on our side, so there wasn't anything to be logged, right?

File: src/utils/email.js [hunk 3]
```diff
@@ -51,40 +51,46 @@ async function initEmail (smtpUrl = AppConstants.SMTP_URL) {
...
+
+    /* c8 ignore next 5 */
+    if (gTransporter.transporter.name === 'JSONTransport') {
+      // @ts-ignore Added typing later, but it disagrees with actual use:
+      console.info('JSONTransport', { message: info.message.toString() })
+      return info;
+    }
+
+    console.info("sent_email", { info });
```
[the remark this answers is not in the document]
Comment (rhelmer) [reply src/utils/email.js:83]: I think it's just metadata but I will confirm before landing: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/425f0ebe145ac9337b0bcbd1084a548d729b4963/types/nodemailer/lib/smtp-connection/index.d.ts#L114-L129

[the remark this answers is not in the document]
Comment (rhelmer) [reply src/utils/email.js:83]: > I didn't see any errors logged locally even though AFAIK emails still don't get sent, but I think the problem there was that it didn't even produce an error on our side, so there wasn't anything to be logged, right?

**Review by mansaj — APPROVED** [review]
LGTM

**Pull request merged**